velero-plugin-for-microsoft-azure/common.go

```diff
@@ -45,6 +45,10 @@ func GetResticEnvVars(config map[string]string) (map[string]string, error) {
 		return nil, err
 	}
 
+	if _, err := getRequiredValues(mapLookup(config), storageAccountConfigKey); err != nil {
```

Comment:
I think we want to remove this, since we don't want to require use of static storage account keys, and the call to `getStorageAccountKey(...)` above should fetch the storage account key, whether it's statically defined in the secret, or gotten through the API.

Reply:
I think that this check should be here as the `config[storageAccountConfigKey]` is accessed a few lines after (velero-plugin-for-microsoft-azure/common.go:53)

Reply:
You're right, I misread this.

However, it turns out that this function isn't even used anymore, and we can just delete it. We're in a slightly messy situation right now, where the code to make restic work on Azure is actually still in the main velero repo (see https://github.com/vmware-tanzu/velero/blob/master/pkg/restic/azure.go), and there's a bunch of duplication between that code and this code.  This happened when we extracted the cloud-provider plugins out of the main velero repo.

So - let's delete the `GetResticEnvVars` function from here. We may also need to follow up with some changes to https://github.com/vmware-tanzu/velero/blob/master/pkg/restic/azure.go to support static storage account keys.

Reply:
The code in https://github.com/vmware-tanzu/velero/blob/master/pkg/restic/azure.go seems like Copy & Paste. Are there any plans to use plugin functionality from the Restic part? If so, this should stay here and be used by Restic.

Reply:
Yeah, it is copy-paste. It's needed both within the `velero` binary to support restic access to Azure, and within the plugin to support all the normal plugin operations. Long-term, we of course want to remove this duplication, but we don't have a concrete plan for it right now.

Reply:
I'd still recommend removing the `GetResticEnvVars` function from here (the plugin), since it isn't actually used here and was inadvertently copied in. But, we will want to make updates to that function in the main velero repo so Azure+restic users can also use static storage account keys.